### ISSUE TITLE
CE2 Breaks color coding when multiple types are defined in JSDoc

### ISSUE THREAD

**Issue by Lukanibal** [YoYoGames/GameMaker-Bugs#8704]
### Description
When defining parameters for functions, using the comma or vertical bar breaks the color coding and it goes back to the default color only after the comma/vertical bar.
![Image](https://github.com/user-attachments/assets/c070c8a6-88f3-470f-9475-99c25dd3de54)
Picture because I'm bad with words.
I also included a package with this script.

### Steps To Reproduce
1. Start GameMaker
2. Define a Function
3. Define a parameter that can accept multiple types
4. See the issue

### Which version of GameMaker are you reporting this issue for?
IDE v2024.1100.0.713 Runtime v2024.1100.0.726

### Which operating system(s) are you seeing the problem on?
Windows 10.0.26100.0

[cc5261ac-625c-4abc-be1c-2018aa1e1e0d](https://api.gamemaker.io/api/github/downloads/cc5261ac-625c-4abc-be1c-2018aa1e1e0d)

**Comment by zreedy** [comment]
Fixed. Additional bug where the `*` in `any*` was not highlighted.

![Image](https://github.com/user-attachments/assets/10681797-06d4-4d71-b1ec-0e9af4bd7a9d)

**Comment by PrismaticRealms** [comment]
Also, it seems the @func doesn't get coloured like the other JSDoc directives.

<!-- Uploading "Screenshot 2024-12-18 at 6.36.24 PM.png"... -→

**Comment by bocoboc** [comment]
> Also, it seems the [@func](https://github.com/func) doesn't get coloured like the other JSDoc directives.

Function directive has been deprecated for CE2 as it's created by the function name and parameters automatically.

**Comment by mgeddesGM** [comment]
verified as of  IDE v2024.1300.0.714  Runtime v2024.1300.0.727

**Comment by PrismaticRealms** [comment]
> > Also, it seems the [@func](https://github.com/func) doesn't get coloured like the other JSDoc directives.
> 
> Function directive has been deprecated for CE2 as it's created by the function name and parameters automatically.

A question about this.

Since the @func directive is deprecated in CE2, how would one define a JSDoc header for a function/method that has a variable number of parameters?  With @func, it would be sometihng like this:

/// @func   MyFunction(param1, param2, param3, ...)

How would we accomplish this without the @func directive in CE2 so that it is obvious in the Feather popup?

**Comment by bocoboc** [comment]
> > > Also, it seems the [@func](https://github.com/func) doesn't get coloured like the other JSDoc directives.
> > 
> > 
> > Function directive has been deprecated for CE2 as it's created by the function name and parameters automatically.
> 
> A question about this.
> 
> Since the [@func](https://github.com/func) directive is deprecated in CE2, how would one define a JSDoc header for a function/method that has a variable number of parameters? With [@func](https://github.com/func), it would be sometihng like this:
> 
> `/// @func MyFunction(param1, param2, param3, ...)`
> 
> How would we accomplish this without the [@func](https://github.com/func) directive in CE2 so that it is obvious in the Feather popup?

You can use an Array as an argument type or use 'Any' as a type, check if the argument is an array, and cycle through it.

```
/// @param {Any} value
function newFunc(_values) {
    if(is_array(_values) {
        for(var i = 0; i < array_length(_values); i++) {
            ...
        }
    }
    else {
        ...
    }
}

or something like

/// @param {Real, Array<Real>} value    -    for type restriction checking
```

**Comment by PrismaticRealms** [comment]
> You can use an Array as an argument type or use 'Any' as a type, check if the argument is an array, and cycle through it.

This requires changing the function signature, which I don't want to do.  Let me ask a different question with the same goal in mind.  How will the JSDocs for the GM runtime function `choose` be defined going forward without using the @func directive such that it is communicated to the programmer that it requires a variable number of parameters as part of it's signature? Image below for reference.

[
![](https://www.prismaticrealms.com/share/GMS2/ChoosePopup.png)
](url)

**Comment by bocoboc** [comment]
I totally understand this. But from what I tried you can't properly replicate the behavior of built-in functions even now.

Feather tooltip already ignores @func, it's just added as a description (that's why I have stopped using it).

![Image](https://github.com/user-attachments/assets/34b52e46-23ac-402e-a2b9-f2b9e1b36730)

But the bottom bar (which is removed in CE2 and we want it back #7688) does show variable argument count when @func is omitted.

![Image](https://github.com/user-attachments/assets/8be078cf-15f2-40b4-851a-0b1dc3a9e94e)

Anyway, it's still ignored by the first line in the tooltip.

![Image](https://github.com/user-attachments/assets/f7fd9e84-fbc4-49cd-8b78-65b9a02c389e)

You could make a new Issue about it.
